[Meta] ROOT's llvm has issues with ranges coming from the Macos 14.3 sdk (llvm 15)

### Check duplicate issues.

- [X] Checked for duplicates

### Description

If the head of ROOT is taken, C++ 20 specified and the build process started, the pcm file of libCore cannot be generated because of this error (at least):
```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__ranges/elements_view.h:155:38: error: cannot take address of consteval function '__get_iterator_category' outside of an immediate invocation
  using iterator_category = decltype(__get_iterator_category());
 ```

### Reproducer

Take a mac with the 14.3 sdk (LLVM 15)root master, enable C++ 20, cmake, make.

### ROOT version

master and 6.28

### Installation method

build from sources

### Operating system

macos

### Additional context

_No response_

--- a/interpreter/llvm/src/tools/clang/lib/Sema/SemaExpr.cpp
+++ b/interpreter/llvm/src/tools/clang/lib/Sema/SemaExpr.cpp
@@ -18763,7 +18763,7 @@ void Sema::MarkDeclRefReferenced(DeclRefExpr *E, const Expr *Base) {
 
   if (auto *FD = dyn_cast<FunctionDecl>(E->getDecl()))
     if (!isConstantEvaluated() && FD->isConsteval() &&
-        !RebuildingImmediateInvocation)
+        !RebuildingImmediateInvocation && !FD->isDependentContext())
       ExprEvalContexts.back().ReferenceToConsteval.insert(E);
   MarkExprReferenced(*this, E->getLocation(), E->getDecl(), E, OdrUse,
                      RefsMinusAssignments);

